### PR TITLE
More verbose warning for multiple network argument error.

### PR DESCRIPTION
**Pull request by amovfx**
This PR has bitcoin print a more verbose message when a network argument collisions happens. 
I'm new to using the bitcoin software and was reconfiguring an old node. I also have limited experience at coding, this is my first PR on an open-source project, so any feedback on how to improve this process is welcomed.

If a chain argument is in the bitcoin.conf and then a bitcoin command has a network flag or a chain argument, such as : bitcoind -regtest  currently you are presented with the message:
```
"Invalid combination of -regtest, -signet, -testnet and -chain. Can use at most one."
```
As I am a new to this software, with not knowing that the bitcoin.conf was being merge in the command line, I was at a loss until I got guidance in the bitcoin-core-dev irc. 

In this PR, if a user makes this error, they will be notified to check their conf files.

The new error is a bit more verbose for the command bitcoind -regtest if the chain argument is set in the command line.
```
"Invalid combination of -regtest, -signet, -testnet and -chain. Can use at most one. Please check if chain is specified in config file."
```

**In regards to the unit test util_tests/util_ChainMerge:**

I had to write a script to check if the resulting output of master branch was the same output as the test from my branch and then update the hash accordingly. It is a simple compare of two txt files.

You can download the scripts here:
https://github.com/amovfx/btc-argument-collision-warning.git

I understand that your time is valuable and this isn't a significant change but I thought it was something within my ability I could contribute. Thank you for your time. 
Also, thanks to everyone that helped me get to this point to even submit a PR. Jonas at chain code, the blockchains commons project, programming bitcoin by Jimmy Song, Achow101's twitch stream, the many hosts of the bitcoin-pr-review club and the devs in the bitcoin-core-dev and bitcoin stack exchange answering my questions.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
What about just modifying the error string to encourage the user to re-check the command line and config file?

**Comment by amovfx** [comment]
Hi @MarcoFalke, Well that would be simpler for sure. I am open to that if you think it is a good idea. Perhaps I went overboard. I can modify that string, ditch the feature test and make sure the unit test still works.

I'm trying to get past this listing error on python -utf8. If you have any insights that would be appreciated.

**Comment by amovfx** [comment]
@MarcoFalke I made your suggested changes, doesn't need the additional functional test. LInting error solved.

File: src/util/system.cpp [hunk 1]
```diff
@@ -1061,7 +1061,7 @@ std::string ArgsManager::GetChainName() const
         LOCK(cs_args);
         util::SettingsValue value = util::GetSetting(m_settings, /* section= */ "", SettingName(arg),
             /* ignore_default_section_config= */ false,
-            /*ignore_nonpersistent=*/false,
+            /* ignore_nonpersistent= */ false,
```
Comment (fanquake) [inline src/util/system.cpp:1064]: You can remove this, and the rest of the unrelated code changes.

File: src/util/system.cpp [hunk 2]
```diff
@@ -1072,16 +1072,14 @@ std::string ArgsManager::GetChainName() const
     const bool is_chain_arg_set = IsArgSet("-chain");
 
     if ((int)is_chain_arg_set + (int)fRegTest + (int)fSigNet + (int)fTestNet > 1) {
-        throw std::runtime_error("Invalid combination of -regtest, -signet, -testnet and -chain. Can use at most one.");
+        throw std::runtime_error("Invalid combination of -regtest, -signet, -testnet and -chain. Can use at most one. Please check if chain is specified in config file.");
     }
     if (fRegTest)
         return CBaseChainParams::REGTEST;
-    if (fSigNet) {
+    if (fSigNet)
```
Comment (luke-jr) [inline src/util/system.cpp:1079]: This is a regression in coding style. If you want to update these, they should probably all be one liners.

eg
```suggestion
    if (fSigNet) return CBaseChainParams::SIGNET;
```

**Review by amovfx — COMMENTED** [review]
Thanks @luke-jr @MarcoFalke 
I'm not sure the protocol for submitting suggested changes. I'm squashing the commit down. Or would it be better to have the suggested changes into a separate commit for better transparency for future reviewers?

**Review by maflcko — APPROVED** [review]

File: src/util/system.cpp [hunk 3]
```diff
@@ -1072,16 +1072,11 @@ std::string ArgsManager::GetChainName() const
     const bool is_chain_arg_set = IsArgSet("-chain");
 
     if ((int)is_chain_arg_set + (int)fRegTest + (int)fSigNet + (int)fTestNet > 1) {
-        throw std::runtime_error("Invalid combination of -regtest, -signet, -testnet and -chain. Can use at most one.");
+        throw std::runtime_error("Invalid combination of -regtest, -signet, -testnet and -chain. Can use at most one. Please check if chain is specified in config file.");
```
Comment (jonatack) [inline src/util/system.cpp:1075]:
```suggestion
        throw std::runtime_error("Invalid combination of -regtest, -signet, -testnet and -chain. Can use at most one. Please check your config options on the command line and in the bitcoin.conf file.");
```

**Comment by jonatack** [comment]
> I'm not sure the protocol for submitting suggested changes. I'm squashing the commit down. Or would it be better to have the suggested changes into a separate commit for better transparency for future reviewers?

Yes, in this project it's generally preferred to squash the changes, if needed, into their final form. See `CONTRIBUTING.md` for more info.

**Comment by fanquake** [comment]
@amovfx are you going to follow up here? This is not currently mergable.
